inference-engine/src/vpu/graph_transformer/src/frontend/detect_network_batch.cpp

```diff
@@ -59,117 +44,91 @@ void FrontEnd::detectNetworkBatch(
     ie::InputsDataMap inputsInfo = network.getInputsInfo();
     ie::OutputsDataMap outputsInfo = network.getOutputsInfo();
 
-    std::map<std::string, ie::Precision> outputsPresisions;
-    for (const auto& pair : outputsInfo)
-        outputsPresisions[pair.first] = pair.second->getPrecision();
-
     //
     // Collect input shapes and remove batch from them.
     //
 
     env.log->trace("Remove batch from inputs");
 
-    ie::ICNNNetwork::InputShapes inputShapes;
+    size_t inputBatch = 0;
 
     for (const auto& p : inputsInfo) {
         VPU_LOGGER_SECTION(env.log);
-
         const auto info = p.second;
         IE_ASSERT(info != nullptr);
 
         const auto ieData = info->getInputData();
         IE_ASSERT(ieData != nullptr);
-
         auto ieShapes = ieData->getTensorDesc().getDims();
         env.log->trace("Input [%s] : %v", p.first, ieShapes);
-
-        switch (ieData->getLayout()) {
-        case ie::Layout::NCDHW:
-        case ie::Layout::NDHWC:
-        case ie::Layout::NCHW:
-        case ie::Layout::NHWC:
-        case ie::Layout::NC:
-        case ie::Layout::CN:
-            ieShapes[0] = 1;
-            break;
-        default:
-            VPU_THROW_FORMAT("Input %v has unexpected layout %v", ieData->getName(), ieData->getLayout());
+        // assume only 4D and 5D inputs have batch
+        if (ieShapes.size() == 4 || ieShapes.size() == 5) {
+            auto batch = ieShapes[0];
+            if (!inputBatch)
+                inputBatch = batch;
+
+            if (inputBatch != batch) {
+                env.log->trace("Network input with name %s has different batch dim then previous inputs, aborting batch detection", p.first);
+                return;
+            }
         }
+    }
+
+    if (inputBatch == 1) {
+        env.log->trace("Network is batch 1. Not need to reshape");
+        return;
+    }
 
+    if (inputBatch == 0) {
+        env.log->trace("Unable to decide on network batch size.");
+        return;
+    }
+    
+    size_t dimsSize = 0;
+    ShapesMap inputShapes;
+    
+    for (const auto& p : inputsInfo) {
+        const auto info = p.second;
+        const auto ieData = info->getInputData();
+        auto ieShapes = ieData->getTensorDesc().getDims();
+        if (!dimsSize)
+            dimsSize = ieShapes.size();
+        if (dimsSize != ieShapes.size())
```

Comment:
Do you accept only inputs with the same number of dims? Why? 
What about a case when the first input is 4D and the second is 2D img_info?